std/array.d

```diff
@@ -2267,7 +2267,16 @@ if (isInputRange!Range &&
             copy(stuff, retval[from .. from + stuff.length]);
 
         retval[from + stuff.length .. $] = subject[to .. $];
-        return cast(T[]) retval;
+        static if (is(T == const) || is(T == immutable))
+        {
+            auto ret = () @trusted { return cast(T[]) retval; } ();
+            retval = null;
+            return ret;
```

Comment:
Why not simply `return ()...`

Reply:
I was copying the behavior of `assumeUnique`

Reply:
Well, `assumeUnique` is very old. It's in a9bc53273c7930c9b2cf442e3d8bc4e28d65ee4b from 2007 since git tracking started and presumably a lot older and I don't see any reason why this is necessary in D2